pkg/utils/utils.go

```diff
@@ -413,3 +419,45 @@ func ConvertToType[T any](obj interface{}) (T, error) {
 
 	return v, nil
 }
+
+// FetchValueFromMetadata fetches a key from a metadata if it exists. It will recursively look in
+// embedded values as well. Must be a unique key, otherwise it will just return the first
+// occurrence.
+func FetchValueFromMetadata[T any](key string, data *apiextensionsv1.JSON, def T) (t T, _ error) {
+	if data == nil {
+		return def, nil
+	}
+
+	m := map[string]any{}
+	if err := json.Unmarshal(data.Raw, &m); err != nil {
+		return t, fmt.Errorf("failed to parse JSON raw data: %w", err)
+	}
+
+	v, err := dig[T](key, m)
+	if err != nil {
+		if errors.Is(err, errKeyNotFound) {
+			return def, nil
+		}
```

Comment:
I believe you forgot to return an error here.

Reply:
The intention here is that if the key is not found we return the set default value. Arguably I could have named the variable better. 🤣 or at least add a comment.

Reply:
Ah, I see what you are trying to do here 👍